pkg/scheduler/framework/plugins/podtopologyspread/filtering.go

```diff
@@ -57,11 +71,15 @@ func (s *preFilterState) Clone() framework.StateData {
 		// Constraints are shared because they don't change.
 		Constraints:          s.Constraints,
 		TpKeyToCriticalPaths: make(map[string]*criticalPaths, len(s.TpKeyToCriticalPaths)),
+		TpKeyToDomainsNum:    make(map[string]int, len(s.TpKeyToDomainsNum)),
```

Comment:
No matter how the preemption strategy looks like, this field is indeed unchanged, so I _think_ we can save efforts to just set s.TpKeyToDomainsNum here.

Reply:
Thanks, I didn't know this. 😓 
>	// clone should make shallow copies for members (e.g., slices or maps) that are not
	// impacted by PreFilter's optional AddPod/RemovePod methods.
https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/cycle_state.go#L31